### PR TITLE
OCT-2240: Initial setup for FactoryBoy

### PR DESCRIPTION
**Pull request by kgarbacinski**
## Definition of Done

1. [ ] If required, the desciption of your change is added to the [QA changelog](https://www.notion.so/octantapp/Changelog-for-the-QA-d96fa3b411cf488bb1d8d9a598d88281) 
2. [ ] Acceptance criteria are met.
3. [ ] PR is manually tested before the merge by developer(s).
    - [ ] Happy path is manually checked.
4. [ ] PR is manually tested by QA when their assistance is required (1).
    - [ ] Octant Areas & Test Cases are checked for impact and updated if required (2).
5. [ ] Unit tests are added unless there is a reason to omit them.
6. [ ] Automated tests are added when required.
7. [ ] The code is merged.
8. [ ] Tech documentation is added / updated, reviewed and approved (including mandatory approval by a code owner, should such exist for changed files).
    - [ ] BE: Swagger documentation is updated.
9. [ ] When required by QA:
    - [ ] Deployed to the relevant environment.
    - [ ] Passed system tests.

---

(1) Developer(s) in coordination with QA decide whether it's required. For small tickets introducing small changes QA assistance is most probably not required.

(2) [Octant Areas & Test Cases](https://docs.google.com/spreadsheets/d/1cRe6dxuKJV3a4ZskAwWEPvrFkQm6rEfyUCYwLTYw_Cc).

### CODE REVIEW

**Review by adam-gf — COMMENTED** [review]
General questions about checksums for addresses and the flexibility of the `setattr` use

File: backend/tests/v2/factories/allocations.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+class AllocationFactory(AsyncSQLAlchemyFactory):
+    class Meta:
+        model = Allocation
+
+    user_id = None
+    epoch = None
+    project_address = LazyAttribute(
+        lambda _: "0x" + "".join(random.choices(string.hexdigits, k=40))
+    )
```
[the remark this answers is not in the document]
Comment (kgarbacinski) [reply backend/tests/v2/factories/allocations.py]: yes, I unified it to use checksum addresses everywhere in the code

File: backend/tests/v2/factories/__init__.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        Associates the provided SQLAlchemy session with the
+        specified factory sets, making them available as attributes of the instance.
+
+
+        Args:
+            fast_session: An SQLAlchemy session to be shared among
+                     the aggregated factory sets.
+        """
+        for attr_name, factory_set in self._factory_sets.items():
+            setattr(self, attr_name, factory_set(fast_session))
```
[the remark this answers is not in the document]
Comment (kgarbacinski) [reply backend/tests/v2/factories/__init__.py:66]: typehints work fine, i double checked it

File: backend/tests/v2/factories/users.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        return user
+
+    async def get_alice(self, session):
+        return await self.get_or_create(session, ALICE_ADDRESS)
+
+    async def get_bob(self, session):
+        return await self.get_or_create(session, BOB_ADDRESS)
+
+    async def get_charlie(self, session):
+        return await self.get_or_create(session, CHARLIE_ADDRESS)
```
[the remark this answers is not in the document]
Comment (kgarbacinski) [reply backend/tests/v2/factories/users.py]: `_meta` is for Factories itself but we pass the `fast_session` from `FactoriesAggregator`, I overlooked that - fixed

File: backend/tests/v2/factories/helpers.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+from web3 import Web3
+
+
+def generate_random_eip55_address() -> ChecksumAddress:
+    """
+    Generate a random checksummed address.
+    """
+    return Web3.to_checksum_address(
+        "0x" + "".join(random.choices(string.hexdigits, k=40)).lower()
+    )
```
Comment (adam-gf) [inline backend/tests/v2/factories/helpers.py:14]: What do you think about using "our" `Address` type (validation on pydantics side via `address = TypeAdapter(Address).validate_python(address)`

Comment (kgarbacinski) [reply backend/tests/v2/factories/helpers.py:15]: changed

**Review by adam-gf — APPROVED** [review]

**Pull request merged**